python/ray/dag/compiled_dag_node.py

```diff
@@ -848,26 +857,39 @@ def _get_or_compile(
             visited.add(cur_idx)
 
             task = self.idx_to_task[cur_idx]
-            # Create an output buffer for the actor method.
-            assert task.output_channel is None
-
             type_hint = task.dag_node.type_hint
             if type_hint.requires_nccl():
                 type_hint.set_nccl_group_id(self._nccl_group_id)
 
             if isinstance(task.dag_node, ClassMethodNode):
+                # Create output buffers for the actor method.
+                assert len(task.output_channels) == 0
                 # `readers` is the nodes that are ordered after the current one (`task`)
                 # in the DAG.
-                readers = [self.idx_to_task[idx] for idx in task.downstream_node_idxs]
-                reader_and_node_list: List[Tuple["ray.actor.ActorHandle", str]] = []
-                dag_nodes = [reader.dag_node for reader in readers]
-                read_by_multi_output_node = False
+                output_to_readers: Dict[CompiledTask, List[CompiledTask]] = {}
+                for idx in task.downstream_node_idxs:
+                    downstream_task = self.idx_to_task[idx]
+                    if not isinstance(downstream_task.dag_node, ClassMethodOutputNode):
+                        if task not in output_to_readers:
+                            output_to_readers[task] = []
+                        output_to_readers[task].append(downstream_task)
+                    else:
+                        output_to_readers[downstream_task] = [
+                            self.idx_to_task[idx]
+                            for idx in downstream_task.downstream_node_idxs
+                        ]
+                output_task_to_reader_and_node_list: Dict[
```

Comment:
Can you add a comment to explain what this data structure is?

Reply:
Done.